src/utils/format_number_si.rs

```diff
@@ -0,0 +1,206 @@
+const SUFFIXES: [&str; 3] = ["k", "M", "B"];
+
+pub fn format_number_si(number: i64) -> String {
+  if number < 1000 {
+    number.to_string()
+  } else {
+    let mut number_as_float = number as f32;
+    for suffix in SUFFIXES {
+      number_as_float /= 1000f32;
+
+      if number_as_float < 1000f32 {
+        return format!(
+          "{:.*}{suffix}",
+          if (number_as_float - number_as_float.floor()) < 0.1 || number_as_float >= 100f32 {
+            0
+          } else {
+            1
+          },
+          number_as_float
+        );
+      }
+    }
+
+    panic!("Number {number} is larger than 1 trillion! Where are you getting numbers this big?");
+  }
+}
```

Comment:
While this can be done with the browser's internationalization API via `js_sys`, that won't work on the server. I also noticed that the old UI was only using the `en-US` locale. If we plan to only handle that locale for these numbers, this algorithm should be fine.

Reply:
There's a library for this also: https://docs.rs/si_format/latest/si_format/

IMO we should just use SI formatting for any potentially large numbers, and not worry about any internationalization on them.